vLLM llama3_subdevices TG test

ppetrovicTT opened the pull request:
### Ticket
https://github.com/tenstorrent/tt-metal/issues/21651

### Problem description
vLLM is not tested on TG with llama3_subdevices

### What's changed
Added the test for vLLM on TG using llama3_subdevices

### Checklist
This is a new test
- [ ✅ ] [vLLM nightly tests](https://github.com/tenstorrent/tt-metal/actions/runs/15250800686)

ppetrovicTT committed:
vLLM TG test

skhorasganiTT commented on .github/workflows/vllm-nightly-tests-impl.yaml:
```diff
@@ -83,15 +93,24 @@ jobs:
           fi
 
-          export HF_MODEL="${{ matrix.test-group.model }}"
+          if [[ "${{ matrix.test-group.runner-label }}" == "config-tg" ]]; then
```
Can this if be an elif?

skhorasganiTT commented on .github/workflows/vllm-nightly-tests-impl.yaml:
```diff
@@ -100,13 +119,15 @@ jobs:
 
       - name: ⏰ Wait for server to be ready
         run: |
+          server_log="./output/vllm_server.log"
```
This is defined a few different times in this file, do you know if it's possible to define once at the beginning?

ppetrovicTT committed:
Improve vLLM test on TG

- Use job env vars for reused filepaths
- Use Llama 3.3 on TG
- Switch-case for different platforms

ttmchiou commented on .github/workflows/vllm-nightly-tests-impl.yaml:
```diff
@@ -13,6 +13,11 @@ on:
         type: string
 
+env:
+  server_pid: "./server.pid"
```
convention dictates this should be capitalized here. nit pick tho

ttmchiou replied:
also can this be located under the ENV variable for container:?

ttmchiou commented on .github/workflows/vllm-nightly-tests-impl.yaml:
```diff
@@ -83,14 +98,24 @@ jobs:
-          export HF_MODEL="${{ matrix.test-group.model }}"
+          case "${{ matrix.test-group.runner-label }}" in
+            "config-t3000")
+              export WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml
```
better to export or set these variables in the matrix.
This will make it cleaner and more readable instead of creating a if statement inside the workflow.

ppetrovicTT replied:
WH_ARCH_YAML and TT_LLAMA_TEXT_VER can't handle empty set, they need to remain unset.
So I'm still doing a check in run, but I guess it's a bit cleaner and it definitely scales better!

ttmchiou commented on .github/workflows/vllm-nightly-tests-impl.yaml:
```diff
@@ -29,6 +34,16 @@ jobs:
+          },
         ]
     runs-on:
       - arch-${{ matrix.test-group.arch }}
```
I just noticed this, maybe i missed this if I reviewed the previous PR. but we shouldn't be using arch-wormhole or any variant of arch-{VAR} as a label.  
TG should use config-tg
T3K should use config-t3000

ppetrovicTT replied:
I followed this as an example, so maybe it's worth fixing it there as well (in another PR):
https://github.com/tenstorrent/tt-metal/blob/main/.github/workflows/t3000-demo-tests-impl.yaml#L43

ppetrovicTT committed:
Reorganize env vars; Print server log on failure

ppetrovicTT committed:
Remove ARCH_NAME

ppetrovicTT committed:
Don't print header when printing server log

ppetrovicTT committed:
Fix override_tt_config json parsing

ppetrovicTT committed:
Proper formatting for override-tt-config
